src/pages/content-edit/schedule-box.js

```diff
@@ -0,0 +1,142 @@
+import m from "mithril";
+import formatDate from "date-fns/format";
+
+import css from "./head.css";
+
+var DEFAULT_START_TIME = "00:00",
+    DEFAULT_END_TIME   = "23:59",
+    DATE_FORMAT = "YYYY-MM-DD",
+    TIME_FORMAT = "HH:mm",
+    TIMESTAMP_FORMAT = "x";
+
+function scheduleStr(side, date, time) {
+    if(!date) {
+        return null;
+    }
+
+    if(!time) {
+        time = side === "start" ? DEFAULT_START_TIME : DEFAULT_END_TIME;
+    }
+
+    return date + " " + time;
+}
+
+function timestampFromStr(str) {
+    return parseInt(formatDate(str, TIMESTAMP_FORMAT), 10);
+}
+
+function timestamp(side, date, time) {
+    var str = scheduleStr(side, date, time);
+
+    return str ? timestampFromStr(str) : null;
+}
+
+
+// All data in and out of the `scheduler` is in the form of timestamps.
+// All transformations to and from the date+time input fields will be
+// handled by this controller.
+
+export function controller(options) {
+    var ctrl = this;
+
+    ctrl.content = options.content;
+
+    ctrl.inputs = null;
+    ctrl.ts = null;
+
+    ctrl.init = function() {
+         ctrl.makeSchedule();
+    };
+
+    ctrl.makeSchedule = function() {
+        var dates = ctrl.content.get().dates,
+            pub   = dates.published_at,
+            unpub = dates.unpublished_at;
+
+        ctrl.inputs = {
+            valid : dates.validSchedule,
+
+            start : {
+                date : pub ? formatDate(pub, DATE_FORMAT) : "",
+                time : pub ? formatDate(pub, TIME_FORMAT) : DEFAULT_START_TIME
+            },
+
+            end : {
+                date : unpub ? formatDate(unpub, DATE_FORMAT) : "",
+                time : unpub ? formatDate(unpub, TIME_FORMAT) : DEFAULT_END_TIME
+            }
+        };
+    };
+
+    ctrl.get = function(side, part) {
```

Comment:
This function is only used once, is very simple, and obscures where the data actually comes from. Remove it please.